tests/petsc/petsc_ts_03.cc

```diff
@@ -168,6 +170,31 @@ class ExponentialDecay
       deallog << "  y =" << sol[0] << "  (exact: " << std::exp(-kappa * t)
               << ')' << std::endl;
     };
+
+    // This callback is invoked after a successfull stage.
+    // Here we only print that the callback is invoked.
+    time_stepper.distribute = [&](const real_type t, VectorType &) -> void {
+      deallog << "Distribute at time " << t << std::endl;
+    };
+
+    // This callback is used to decide to remesh.
+    time_stepper.prepare_for_coarsening_and_refinement =
+      [&](const real_type    t,
+          const unsigned int step,
+          const VectorType &,
+          bool &resize) -> void {
+      deallog << "Prepare at time " << t << " and step " << step << std::endl;
+      resize = (step && step % 5 == 0);
+    };
```

Comment:
This will then become the following:
```suggestion
    time_stepper.prepare_for_coarsening_and_refinement =
      [&](const real_type    t,
          const unsigned int step,
          const VectorType &) -> bool {
      deallog << "Prepare at time " << t << " and step " << step << std::endl;
      return (step && step % 5 == 0);
    };
```

Reply:
if I return a `bool` instead of of having a `void` function I cannot use the `call_and_possibly_..` wrapping